cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -112,39 +116,42 @@ def batch_prompt(
                         option_dict,
                         retries=retries,
                     )
-                    for prompt, option_dict in zip(prompts, options)
+                    for prompt, option_dict in zip(prompts, options_collection)
                 ],
                 timeout=timeout,
             )
         )
 
         return cast(List[TLMResponse], tlm_responses)
 
-    def batch_get_confidence_score(
+    def _batch_get_confidence_score(
         self,
-        prompts: List[str],
-        responses: List[str],
-        options: Union[None, TLMOptions, List[Union[TLMOptions, None]]] = None,
+        prompts: Collection[str],
+        responses: Collection[str],
+        options: Union[None, TLMOptions, Collection[Union[TLMOptions, None]]] = None,
         timeout: Optional[float] = None,
-        retries: int = 0,
+        retries: int = 1,
     ) -> List[float]:
         """Run batch of TLM get confidence score.
 
         Args:
-            prompts (List[str]): list of prompts to run get confidence score for
-            responses (List[str]): list of responses to run get confidence score for
+            prompts (Collection[str]): list or other iterable of prompts to run get confidence score for
+            responses (Collection[str]): list or other iterable of responses to run get confidence score for
             options (None | TLMOptions | List[TLMOptions  |  None], optional): list of options (or instance of options) to pass to get confidence score method. Defaults to None.
             timeout (Optional[float], optional): timeout (in seconds) to run all prompts. Defaults to None.
-            retries (int): number of retries to attempt for each individual prompt. Defaults to 0.
+            retries (int): number of retries to attempt for each individual prompt in case of error. Defaults to 1.
 
         Returns:
             List[float]: TLM confidence score for each prompt (in supplied order)
         """
-        if not isinstance(options, list):
-            options = [options for _ in prompts]
+        if is_collection(options):
+            options_collection = cast(Collection[Union[TLMOptions, None]], options)
+        else:
+            options = cast(Union[None, TLMOptions], options)
+            options_collection = [options for _ in prompts]
 
         assert len(prompts) == len(responses), "Length of prompts and responses must match."
```

Comment:
```python3
def is_collection(obj: object) -> bool:
    return hasattr(obj, "__iter__") and hasattr(obj, "__len__")

string_prompt = "<your prompt>"
is_collection(string_prompt)
```

Code above returns True. This becomes an issue at this line if two strings are passed since the len() of the strings does not match. This should only be checked for lists.

Reply:
fixed in latest commit

Reply:
we don't necessarily want to limit this to just lists. this (as implemented) could take in a list, tuple, np array, etc.